vmdb/app/models/ext_management_system.rb

```diff
@@ -27,7 +27,9 @@ def self.supported_subclasses
 
   def self.supported_types_and_descriptions_hash
     supported_subclasses.each_with_object({}) do |klass, hash|
-      hash[klass.ems_type] = klass.description
+      if Vmdb::PermissionStores.instance.supported_ems_type?(klass.ems_type)
+        hash[klass.ems_type] = klass.description
+      end
```

Comment:
This should be in the supported_subclasses method instead, and then this guy will just work itself out.